src/ScottPlot5/ScottPlot/DataSource/SignalArray.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace ScottPlot.DataSource;
+
+/// <summary>
+/// This data source has evenly-spaced Xs and Ys stored in a fixed-length array.
+/// Changes made to the contents of the array will appear when the plot is rendered.
+/// </summary>
+public class SignalArray : IDataSource, ISignalSource
+{
+    public double Period { get; set; }
+    public double XOffset { get; set; }
+    private readonly double[] Ys;
+
+    public int Count => Ys.Length;
+
+    public SignalArray(double[] ys, double period)
+    {
+        Ys = ys;
+        Period = period;
+    }
+
+    public CoordinateRange GetMinMax(CoordinateRange xRange)
+    {
+        throw new NotImplementedException();
+    }
+
+    public int GetIndex(double x)
+    {
+        return (int)((x - XOffset) / Period);
+    }
+
+    public double GetX(int index)
+    {
+        return (index * Period) + XOffset;
+    }
+
+    public Coordinates this[int index]
+    {
+        get => new(GetX(index), Ys[index]);
+        set { Ys[index] = value.Y; }
```

Comment:
no, but also this seems like a weird way to iterate over signal values... I'm thinking of removing `List<Coordinates>` because it's pretty inefficient to loop this way. 

Maybe we should totally break apart the two main types of data sources: scatter-type and signal-type

I'll keep going and see how this evolves